s3tests_boto3/functional/test_iam.py

```diff
@@ -413,3 +419,219 @@ def test_delete_user_policy_from_multiple_policies():
+        if bucket == response['Buckets'][index]['Name']:
+            bucket_listed = True
+            break
+        else:
+            continue
+    if not bucket_listed:
+        raise AssertionError("bucket is not listed")
+
+    s3_client.delete_bucket(Bucket=bucket)
+    response = s3_client.list_buckets()
+    bucket_listed = False
+    for index in range(len(response['Buckets'])):
+        if bucket == response['Buckets'][index]['Name']:
+            bucket_listed = True
+            break
+        else:
+            continue
+    if bucket_listed:
+        raise AssertionError("deleted bucket is getting listed")
+    client.delete_user_policy(PolicyName='AllowAccessPolicy', UserName=get_tenant_user_id())
+
+
+@attr(resource='user-policy')
+@attr(method='s3 Actions')
+@attr(operation='Verify Deny Bucket Actions in user Policy')
+@attr(assertion='succeeds')
+@attr('user-policy')
+def test_deny_bucket_actions_in_user_policy():
+    client = get_tenant_iam_client()
+    bucket = get_new_bucket()
+    policy_document_allow = json.dumps(
+        {"Version": "2012-10-17",
+         "Statement": {
+             "Effect": "Deny",
+             "Action": ["s3:ListAllMyBuckets", "s3:DeleteBucket"],
+             "Resource": "arn:aws:s3:::*"}}
+    )
+
+    client.put_user_policy(PolicyDocument=policy_document_allow, PolicyName='DenyAccessPolicy',
+                           UserName=get_tenant_user_id())
+    s3_client = get_client()
+
+    e = assert_raises(ClientError, s3_client.list_buckets, Bucket=bucket)
+    status, error_code = _get_status_and_error_code(e.response)
+    eq(status, 403)
+    eq(error_code, 'AccessDenied')
+    e = assert_raises(ClientError, s3_client.delete_bucket, Bucket=bucket)
+    status, error_code = _get_status_and_error_code(e.response)
+    eq(status, 403)
+    eq(error_code, 'AccessDenied')
+    client.delete_user_policy(PolicyName='DenyAccessPolicy', UserName=get_tenant_user_id())
+    s3_client.delete_bucket(Bucket=bucket)
+
+
+@attr(resource='user-policy')
+@attr(method='s3 Actions')
+@attr(operation='Verify Allow Object Actions in user Policy')
+@attr(assertion='succeeds')
+@attr('user-policy')
+def test_allow_object_actions_in_user_policy():
+    client = get_tenant_iam_client()
+    bucket = get_new_bucket()
+
+    policy_document_allow = json.dumps(
+        {"Version": "2012-10-17",
+         "Statement": {
+             "Effect": "Allow",
+             "Action": ["s3:PutObject", "s3:GetObject", "s3:DeleteObject"],
+             "Resource": f"arn:aws:s3:::{bucket}/*"}}
+    )
+    client.put_user_policy(PolicyDocument=policy_document_allow, PolicyName='AllowAccessPolicy',
+                           UserName=get_tenant_user_id())
+
+    s3_client = get_client()
+    s3_client.put_object(Bucket=bucket, Key='foo', Body='bar')
+    response = s3_client.get_object(Bucket=bucket, Key='foo')
+    body = response['Body'].read()
+    if type(body) is bytes:
+        body = body.decode()
+        print("decode")
```

Comment:
Maybe we can remove this print statement

Reply:
yes, Thank you.